packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -2469,19 +2462,17 @@ export default (superClass: Class<Parser>): Class<Parser> =>
             return this.finishCallExpression(node, state.optionalChainMember);
           }
 
-          // TODO: This doesn't exactly match what TS does when it comes to ASI.
-          // For example,
-          //   a<b>
-          //   if (0);
-          // is not valid TS code (https://github.com/microsoft/TypeScript/issues/48654)
-          // However, it should correctly parse anything that is correctly parsed by TS.
+          const tokenType = this.state.type;
           if (
-            tsTokenCanStartExpression(this.state.type) &&
-            this.state.type !== tt.parenL
+            // a<b>>c is not (a<b>)>c, but a<(b>>c)
+            tokenType === tt.gt ||
+            // a<b>c is (a<b)>c
+            (tokenType !== tt.parenL &&
+              tokenCanStartExpression(tokenType) &&
+              !this.hasPrecedingLineBreak())
```

Comment:
We also have this test (still passing) which is relevant: https://github.com/babel/babel/tree/main/packages/babel-parser/test/fixtures/typescript/type-arguments/call-newline